app/javascript/packages/document-capture/components/in-person-location-post-office-search-step.tsx

```diff
@@ -57,31 +61,34 @@ function InPersonLocationPostOfficeSearchStep({ onChange, toPreviousStep, regist
         }, 250);
         return;
       }
-      // prevent navigation from continuing
-      e.preventDefault();
+
       if (inProgress) {
         return;
       }
+
       const selected = transformKeys(selectedLocation, snakeCase);
       setInProgress(true);
       try {
         await request(LOCATIONS_URL, {
           json: selected,
           method: 'PUT',
         });
+
         if (mountedRef.current) {
           setAutoSubmit(true);
           setImmediate(() => {
             // continue with navigation
             e.target.disabled = false;
             e.target.click();
+
             // allow process to be re-triggered in case submission did not work as expected
             setAutoSubmit(false);
           });
         }
       } finally {
         if (mountedRef.current) {
           setInProgress(false);
+          forceRedirect(inPersonURL!);
```

Comment:
This `finally` is here to ensure the situation is recoverable (i.e. user can click button again if needed). The user shouldn't be redirected though unless the action is successful.

Reply:
@NavaTim would you suggest putting the forceRedirect call in the try block? Or where else?

Reply:
The call to `forceRedirect` probably needs to go in the `setImmediate` callback, replacing the line `e.target.click()`.